spec/jobs/petition_pdf_generation_worker_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe PetitionPdfGenerationWorker do
+  let(:petition_pdf_service) { spy }
```

Comment:
Whenever possible use the `instance_spy` and `class_spy`. RSpec will check if the called methods exists on the spy, which really, really helps.

```instance_spy MyService```